classes/RightSignature/HttpClient.php

```diff
@@ -2,81 +2,154 @@
 
 namespace RightSignature;
 
+use GuzzleHttp\Client;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\RequestException;
+
 /**
  * Communicates with the RightSignature HTTP endpoint.
  */
-class HttpClient
+class HttpClient implements HttpClientInterface
 {
+	/**
+	 * @var ClientInterface
+     */
 	private $_client;
 
-	public static function forToken($apiToken)
+	/**
+	 * @var api token
+     */
+	private $_apiToken;
+
+	/**
+	 * @param The $apiToken
+	 * @param \GuzzleHttp\ClientInterface|null $client
+	 * @return HttpClient
+     */
+	public static function forToken($apiToken, ClientInterface $client = null)
 	{
-		$instance = new self();
-		$instance->_client->addHeader(sprintf('Api-Token: %s', $apiToken));
+		if (! $client) {
+			$client = new Client([
+				'base_uri' => \RightSignature::API_ENDPOINT,
+			]);
+		}
+		$instance = new self($client);
+		$instance->setToken($apiToken);
+
 		return $instance;
 	}
 
-	public function __construct()
+	/**
+	 * @param \GuzzleHttp\ClientInterface $client
+     */
+	private function __construct(ClientInterface $client)
 	{
-		$this->_client = new \Ergo\Http\Client(\RightSignature::API_ENDPOINT);
-		$this->_client->addHeader(sprintf('Api-Version: %s', \RightSignature::API_VERSION));
+		$this->_client = $client;
 	}
 
 	/**
 	 * @param string $path
 	 * @return string
-	 * @throws RightSignature\Exception
+	 * @throws Exception
 	 */
 	public function get($path)
 	{
-		$response = $this->_submit('get', $path);
-		return $response;
+		$getRequest = [
+			$path, [
+				'headers' => [
+					'Api-Token' => $this->_apiToken,
+					'Api-Version' => \RightSignature::API_VERSION
+				],
+				'verify' => false
```

Comment:
I guess this was mainly so that I could use it from a non-ssl host but I see your concern.  Without it the method will fail if used from a non-SSL host.  Should I still take it out?